tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,345 @@
+
+### Detailed Customer Experience
+ <!-- <List of steps the customer goes through from the start to the end of the scenario to provide more detailed view of exactly what the customer is able to do given the new capabilities>  -->
+<!-- Step 1
+Step 2
+… -->
+
+**Step 1: Define and deploy Radius Environment and Recipes**
+
+As an operator, I need to define and deploy a Radius Environment and associate relevant Recipes to that environment so that my development team can build and deploy their Radius applications.
+
+1. Using my editor of choice, such as VSCode, I open an existing Radius environment file, `env.bicep`, from the online Radius samples repo. 
+1. I use Radius documentation to determine specific edits to my `env.bicep` file such as Kubernetes cluster and the appropriate cloud provider registrations. 
+   1. I will create a Radius Environment resource and add all the information listed above. 
+   1. I will then edit my Radius Environment resource to contain the properties required to register my Radius Recipe such as template path, template kind, etc.
+1. I then push my new `env.bicep` file to my git repository, as I would any other file. 
+1. This triggers Flux or ArgoCD to detect the change, read the new `env.bicep` file, and deploy the Radius Environment as specified.
+1. After Flux or ArgoCD initiates the Radius environment deployment, as I would normally do, I use Flux or ArgoCD to monitor deployment status and health status of the underlying Kubernetes cluster. 
+
+> As an operator, while Radius presents some new concepts, those concepts fit cleanly within my existing GitOps workflows so Radius is relatively low overhead for me to adopt and to provide value to my development counterparts.   
+
+**Step 2: Deploy and manage Radius Applications and Resources through GitOps toolset**
+
+Now, as a developer, I need to make some changes to my application code. Specifically, I need to change the Radius Resource for my frontend UI. My frontend resource is already running in my Radius application but I need update the code for parsing user input. First, I'll update the required file where this function lives then I'll update my Radius Resource as follows: 
+
+1. With my editor of choice, such as VSCode, I open the existing `frontend` application file `helper.ts` and update my function `parseUsers` with the new required logic. 
+2. Then I push the changes to my git repository. 
+3. This triggers Flux/ArgoCD to detect the change and update the `frontend` resource with the new function. I use Flux/ArgoCD to monitor the health status of the Kubernetes cluster and immediately realize the pod associated with my `frontend` resource is failing.
+4. I review my code change again and realize I created a bug in the function which is causing the resource to fail. 
+5. To resolve, I:
+   1. Use the Flux/ArgoCD rollback commands to restart my `frontend` resource pods with the previous code version so as not to disrupt my customers. 
```

Comment:
This isn't very gitops-ey - wouldn't I do a `git revert` and push instead?

Reply:
so it looks like Flux users would leverage `git revert` (https://github.com/fluxcd/flux2/discussions/2916#discussioncomment-3205923) but ArgoCD provides a rollback mechanism (https://argo-cd.readthedocs.io/en/release-2.7/user-guide/commands/argocd_app_rollback/)

thus, i'll include details for both.